json_dump/json_dump.py

```diff
@@ -1,91 +1,146 @@
-            print("Message: {0}".format(rec))
-        # Print it to file or stdout
-        file.write(bytes(json.dumps(rec, indent=options.indent) + '\n', "utf-8"))
-        if not options.noflush:
-            file.flush()
-    except ValueError as e:
-        sys.stderr.write(str(e) + '\n')
+import socket
+import sys
+import time
+from typing import Optional
+
+import pytrap
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """Prepare the argument parser.
+
+    Returns:
+         An instance of ArgumentParser with ready arguments.
+    """
+    parser = argparse.ArgumentParser(add_help=True)
+    parser.add_argument("-i", "--ifcspec", dest="ifcspec", metavar="IFCSPEC",
+                        required=True,
+                        help="See https://nemea.liberouter.org/trap-ifcspec/")
+    parser.add_argument("-I", "--indent", metavar="N", type=int,
+                        help="Pretty-print JSON with indentation set to N spaces. "
+                             "Note that such format can't be read by json_replay module.")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Set verbose mode (print messages).")
+    parser.add_argument("--noflush", action="store_true",
+                        help="Disable automatic flush of output buffer after writing a "
+                             "record (may improve performance).")
+
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("-w", dest="filename", metavar="FILE",
+                       help="Write dump to FILE instead of stdout (overwrite file)")
+    group.add_argument("-a", dest="filename_append", metavar="FILE",
+                       help="Write dump to FILE instead of stdout (append to file)")
+    group.add_argument("-s", dest="networktarget", metavar="HOST:PORT",
+                       help="Write dump using a TCP network stream to HOST:PORT")
+    return parser
+
+
+def connect_socket(address: str, port: int) -> socket.SocketIO:
+    """Create a connection to a socket given an address and a port.
+
+    The connection is tried repeatedly until successful. If the connection to the socket
+    fails, wait 5 seconds and retry.
+
+    Args:
+        address (str): The address of the destination socket.
+        port (int): The port of the destination socket.
+
+    Returns:
+        The SocketIO object.
+    """
+    while True:
+        try:
+            s = socket.create_connection((address, port))
+            sys.stderr.write(time.strftime("%F-%T") +
+                             f" Connection to {address}:{port} established.\n")
+            sys.stderr.flush()
+            return socket.SocketIO(s, "w")
+        except socket.error:
+            # sleep for a while and then reconnect
+            sys.stderr.write(time.strftime("%F-%T") +
+                             " Connection failed, trying again in 5 seconds.\n")
+            sys.stderr.flush()
+            time.sleep(5)
```

Comment:
Using magic numbers is usually not a good idea. Use default param instead for better function desing.

```suggestion
def connect_socket(address: str, port: int, wait_interval = 5) -> socket.SocketIO:
    """Create a connection to a socket given an address and a port.

    The connection is tried repeatedly until successful. If the connection to the socket
    fails, wait `wait_interval` seconds and retry.

    Args:
        address (str): The address of the destination socket.
        port (int): The port of the destination socket.
        wait_interval (int): Waiting interval in sec (default value = 5 sec)

    Returns:
        The SocketIO object.
    """
    while True:
        try:
            s = socket.create_connection((address, port))
            sys.stderr.write(time.strftime("%F-%T") +
                             f" Connection to {address}:{port} established.\n")
            sys.stderr.flush()
            return socket.SocketIO(s, "w")
        except socket.error:
            # sleep for a while and then reconnect
            sys.stderr.write(time.strftime("%F-%T") +
                             f" Connection failed, trying again in {wait_interval} seconds.\n")
            sys.stderr.flush()
            time.sleep(wait_interval)
```